### PR TITLE
add docker file

### PR DESCRIPTION
**Pull request by alexcote1**

### CODE REVIEW

**Comment by Revadike** [comment]
I never used docker. It would be nice to have another reviewer share his/her input on this, before I merge it.

**Comment by Walrus98** [comment]
Docker would be great for this project!

File: Dockerfile
Comment (nstrelow) [inline Dockerfile:4]: You should be able to use
`npm ci --production`

Installs packages from package-lock.json, does not cache and --production does not install devDependencies.

File: Dockerfile
Comment (nstrelow) [inline Dockerfile:1]: Why not use the latest image? And alpine.
Would suggest:
`FROM node:alpine`

File: Dockerfile
Comment (nstrelow) [inline Dockerfile:6]: Why wait? Never used this before

File: startup.sh
Comment (nstrelow) [inline startup.sh:2]: Why sleep then?

**Review by nstrelow — CHANGES_REQUESTED** [review]
Added some comments 😁

File: Dockerfile
Comment (nstrelow) [reply Dockerfile:4]: @Revadike This needs a package-lock.json though.

File: Dockerfile
Comment (LionZXY) [reply Dockerfile:1]: This is bad practice use non-specific version in Dockerfiles

**Comment by LionZXY** [comment]
It really work? How you run npm install without python?

**Comment by Revadike** [comment]
With node.js

**Comment by LionZXY** [comment]
So... This Dockerfiles not work:
```
Error: Failed to launch the browser process!
/usr/src/app/node_modules/puppeteer/.local-chromium/linux-722234/chrome-linux/chrome: error while loading shared libraries: libX11-xcb.so.1: cannot open shared object file: No such file or directory
```

File: Dockerfile
Comment (Revadike) [reply Dockerfile:4]: > @Revadike This needs a package-lock.json though.

Interesting. Perhaps make this PR edit .gitignore to remove `package-lock.json` in it.

**Comment by Revadike** [comment]
Thanks for the reviews. Turning this into a draft PR, until its fully working and reviewed.

File: Dockerfile
Comment (nstrelow) [reply Dockerfile:1]: Makes sense, maybe use the latest working one then (or where they will be the least compatibility problems down the road)

**Comment by Revadike** [comment]
Superseeded by https://github.com/Revadike/epicgames-freebies-claimer/pull/105

**Pull request closed without merging**
